src/vs/workbench/services/extensionManagement/common/extensionManagementService.ts

```diff
@@ -471,35 +473,52 @@ export class ExtensionManagementService extends Disposable implements IWorkbench
 		}
 
 		const limitedSupportMessage = localize('limited support', "'{0}' has limited functionality in {1}.", extension.displayName || extension.identifier.id, productName);
-		let message: string, buttons: string[], detail: string | undefined;
+		let message: string;
+		let buttons: IPromptButton<void>[] = [];
+		let detail: string | undefined;
+
+		const installAnywayButton: IPromptButton<void> = {
+			label: localize({ key: 'install anyways', comment: ['&& denotes a mnemonic'] }, "&&Install Anyway"),
+			run: () => { }
+		};
+
+		const showExtensionsButton: IPromptButton<void> = {
+			label: localize({ key: 'showExtensions', comment: ['&& denotes a mnemonic'] }, "&&Show Extensions"),
+			run: () => this.instantiationService.invokeFunction(accessor => accessor.get(ICommandService).executeCommand('extension.open', extension.identifier.id, 'extensionPack'))
+		};
 
 		if (nonWebExtensions.length && hasLimitedSupport) {
 			message = limitedSupportMessage;
 			detail = `${virtualWorkspaceSupportReason ? `${virtualWorkspaceSupportReason}\n` : ''}${localize('non web extensions detail', "Contains extensions which are not supported.")}`;
-			buttons = [localize('install anyways', "Install Anyway"), localize('showExtensions', "Show Extensions"), localize('cancel', "Cancel")];
+			buttons = [
+				installAnywayButton,
+				showExtensionsButton
+			];
 		}
 
 		else if (hasLimitedSupport) {
 			message = limitedSupportMessage;
 			detail = virtualWorkspaceSupportReason || undefined;
-			buttons = [localize('install anyways', "Install Anyway"), localize('cancel', "Cancel")];
+			buttons = [installAnywayButton];
 		}
 
 		else {
 			message = localize('non web extensions', "'{0}' contains extensions which are not supported in {1}.", extension.displayName || extension.identifier.id, productName);
-			buttons = [localize('install anyways', "Install Anyway"), localize('showExtensions', "Show Extensions"), localize('cancel', "Cancel")];
-		}
-
-		const { choice } = await this.dialogService.show(Severity.Info, message, buttons, { cancelId: buttons.length - 1, detail });
-		if (choice === 0) {
-			return;
-		}
-		if (choice === buttons.length - 2) {
-			// Unfortunately ICommandService cannot be used directly due to cyclic dependencies
-			this.instantiationService.invokeFunction(accessor => accessor.get(ICommandService).executeCommand('extension.open', extension.identifier.id, 'extensionPack'));
-		}
-		throw new CancellationError();
-
+			buttons = [
+				installAnywayButton,
+				showExtensionsButton
+			];
+		}
+
+		await this.dialogService.prompt({
```

Comment:
@sandy081 fyi please check, this one is a bit more complex adoption of the new dialog api